### ISSUE TITLE
Code Health: Minor vet and lint Issues from Code Merger

### ISSUE THREAD

**Issue by matttproud** [golang/mock#104]
vet: gomock/call.go line 149
lint: gomock/call.go line 207

lint mockgen/tests/generated_identifier_conflict/bugreport.go line 5

lint gomock/mockgen/tests/generated_identifier_conflict/bugreport_test.go line 4

Filing for posterity.

**Comment by codyoss** [comment]
Thanks for the report!

**Comment by jesselang** [comment]
Hey @codyoss. I saw the "good first issue" label, and thought I'd have a look. What's the work to be done here? Here's what I see when I run `go vet` and `golint` against `master` branch today:

```
% go version
go version go1.13.4 linux/amd64
% go vet ./...
% golint ./...
gomock/call.go:385:21: error strings should not be capitalized or end with punctuation or a newline
mockgen/internal/tests/aux_imports_embedded_interface/bugreport.go:16:1: exported function CallForeignMethod should have comment or be unexported
mockgen/internal/tests/aux_imports_embedded_interface/faux/faux.go:3:6: exported type Foreign should have comment or be unexported
mockgen/internal/tests/aux_imports_embedded_interface/faux/faux.go:8:6: exported type Embedded should have comment or be unexported
mockgen/internal/tests/aux_imports_embedded_interface/faux/faux.go:10:6: exported type Return should have comment or be unexported
mockgen/internal/tests/copyright_file/input.go:3:1: don't use an underscore in package name
mockgen/internal/tests/copyright_file/input.go:5:6: exported type Empty should have comment or be unexported
mockgen/internal/tests/custom_package_name/client/v1/client.go:5:6: exported type Client should have comment or be unexported
mockgen/internal/tests/custom_package_name/client/v1/client.go:7:1: exported method Client.Greet should have comment or be unexported
mockgen/internal/tests/custom_package_name/client/v1/client.go:11:6: exported type GreetInput should have comment or be unexported
mockgen/internal/tests/custom_package_name/greeter/greeter.go:16:6: exported type InputMaker should have comment or be unexported
mockgen/internal/tests/custom_package_name/greeter/greeter.go:20:6: exported type Greeter should have comment or be unexported
mockgen/internal/tests/custom_package_name/greeter/greeter.go:25:1: exported method Greeter.Greet should have comment or be unexported
mockgen/internal/tests/custom_package_name/validator/validate.go:3:1: exported function Validate should have comment or be unexported
mockgen/internal/tests/dot_imports/input.go:1:1: package comment should be of the form "Package dot_imports ..."
mockgen/internal/tests/dot_imports/input.go:2:1: don't use an underscore in package name
mockgen/internal/tests/dot_imports/input.go:6:2: should not use dot imports
mockgen/internal/tests/dot_imports/input.go:7:2: should not use dot imports
mockgen/internal/tests/dot_imports/input.go:10:6: exported type WithDotImports should have comment or be unexported
mockgen/internal/tests/empty_interface/input.go:1:1: package comment should be of the form "Package empty_interface ..."
mockgen/internal/tests/empty_interface/input.go:2:1: don't use an underscore in package name
mockgen/internal/tests/empty_interface/input.go:4:6: exported type Empty should have comment or be unexported
mockgen/internal/tests/generated_identifier_conflict/bugreport.go:5:6: exported type Example should have comment or be unexported
mockgen/internal/tests/import_source/definition/source.go:1:1: package comment should be of the form "Package source ..."
mockgen/internal/tests/import_source/definition/source.go:5:6: exported type X should have comment or be unexported
mockgen/internal/tests/import_source/definition/source.go:7:6: exported type S should have comment or be unexported
mockgen/internal/tests/internal_pkg/generate.go:1:1: package comment should be of the form "Package test ..."
mockgen/internal/tests/internal_pkg/subdir/internal/pkg/input.go:3:6: exported type Arg should have comment or be unexported
mockgen/internal/tests/internal_pkg/subdir/internal/pkg/input.go:7:6: exported type Intf should have comment or be unexported
mockgen/internal/tests/mock_in_test_package/user.go:5:6: exported type User should have comment or be unexported
mockgen/internal/tests/mock_in_test_package/user.go:9:6: exported type Finder should have comment or be unexported
mockgen/internal/tests/vendor_dep/doc.go:1:1: don't use an underscore in package name
mockgen/internal/tests/vendor_dep/vendor_dep.go:1:1: don't use an underscore in package name
mockgen/internal/tests/vendor_dep/vendor_dep.go:5:6: exported type VendorsDep should have comment or be unexported
mockgen/internal/tests/vendor_pkg/doc.go:1:1: don't use an underscore in package name
mockgen/model/model.go:36:1: exported method Package.Print should have comment or be unexported
mockgen/model/model.go:58:1: exported method Interface.Print should have comment or be unexported
mockgen/model/model.go:78:1: exported method Method.Print should have comment or be unexported
mockgen/model/model.go:116:1: exported method Parameter.Print should have comment or be unexported
mockgen/model/model.go:187:2: exported const RecvDir should have comment (or a comment on this block) or be unexported
mockgen/model/model.go:258:9: if block ends with a return statement, so drop this else and outdent its block
mockgen/model/model.go:286:1: exported function InterfaceFromInterfaceType should have comment or be unexported
sample/user.go:3:1: package comment should be of the form "Package user ..."
sample/user.go:24:2: should not use dot imports
sample/user.go:28:1: comment on exported type Index should be of the form "Index ..." (with optional leading article)
sample/user.go:79:1: comment on exported type Embed should be of the form "Embed ..." (with optional leading article)
sample/user.go:86:6: exported type Embedded should have comment or be unexported
sample/user.go:93:1: comment on exported function Remember should be of the form "Remember ..."
sample/user.go:110:1: exported function GrabPointer should have comment or be unexported
sample/concurrent/concurrent.go:6:6: exported type Math should have comment or be unexported
sample/imp1/imp1.go:5:6: exported type Imp1 should have comment or be unexported
sample/imp1/imp1.go:7:6: exported type ImpT should have comment or be unexported
sample/imp1/imp1.go:9:6: exported type ForeignEmbedded should have comment or be unexported
sample/imp2/imp2.go:3:6: exported type Imp2 should have comment or be unexported
sample/imp3/imp3.go:3:6: exported type Imp3 should have comment or be unexported
sample/imp4/imp4.go:1:1: don't use an underscore in package name
sample/imp4/imp4.go:3:6: exported type Imp4 should have comment or be unexported
```

**Comment by codyoss** [comment]
@jesselang thanks for your interest! I would say for now we can ignore anything in internal and sample, as I don't think the vet reports matter there as much and I would eventually like to get to some refactoring in that package. Other than that I would say the rest are fair game to update.

**Comment by jesselang** [comment]
Solved in #345. I think this can be closed.
